docs/src/main/asciidoc/security-openid-connect-multitenancy.adoc

```diff
@@ -844,6 +844,81 @@ The default tenant configuration is automatically disabled when `quarkus.oidc.au
 
+== Authentication outside of the HTTP request context
+
+Sometimes it is practical to be able to create `SecurityIdentity` for given token when there is no active request context.
+The `quarkus-oidc` extension provides `io.quarkus.oidc.TenantIdentityProvider` to convert a token to a `SecurityIdentity` instance.
+When the provider is used during an HTTP request, the tenant configuration can be resolved as described in the <<static-tenant-resolution>> section.
+However, when there is no active HTTP request, you need to select tenant explicitly with the `io.quarkus.oidc.TenantFeature` qualifier.
+One situation when you need to select the tenant configuration this way is when you are processing messages with the xref:vertx.adoc#event-bus[Vert.x event bus].
+In the example below, the 'product-order' message is consumed within different CDI request context, therefore injected `SecurityIdentity` would be anonymous.
+
+[source,java]
+----
+package org.acme.quickstart.oidc;
+
+import static jakarta.ws.rs.core.HttpHeaders.AUTHORIZATION;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import io.vertx.core.eventbus.EventBus;
+
+@Path("order")
+public class OrderResource {
+
+    @Inject
+    EventBus eventBus;
+
+    @POST
+    public void order(String product, @HeaderParam(AUTHORIZATION) String bearer) {
+        String rawToken = bearer.substring("Bearer ".length()); <1>
+        eventBus.publish("product-order", new Product(product, 1, rawToken));
+    }
+}
+----
+<1> At this point, token is not verified when proactive authentication is disabled.
+
+[source,java]
+----
+package org.acme.quickstart.oidc;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+import io.quarkus.oidc.AccessTokenCredential;
+import io.quarkus.oidc.TenantFeature;
+import io.quarkus.oidc.TenantIdentityProvider;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.vertx.ConsumeEvent;
+import io.smallrye.common.annotation.Blocking;
+
+@ApplicationScoped
+public class OrderService {
+
+    @TenantFeature("tenantId")
+    @Inject
+    TenantIdentityProvider identityProvider;
+
+    @Inject
+    TenantIdentityProvider defaultIdentityProvider; <1>
+
+    @Blocking
+    @ConsumeEvent("product-order")
+    void processOrder(Product product) {
+        String rawToken = product.customerAccessToken;
+        AccessTokenCredential token = new AccessTokenCredential(rawToken);
+        SecurityIdentity = identityProvider.authenticate(token).await().indefinitely(); <2>
+        ...
+    }
+
+}
+----
+<1> For default tenant, the `TenantFeature` qualifier is optional.
+<2> Executes token verification and converts the token to a `SecurityIdentity`.
+
+WARNING: <<tenant-config-resolver>> is currently not supported. Authentication that requires dynamic tenant will fail.
+
 == References
```

Comment:
Hi @michalvavrik Example looks very practical, thanks.

The only I've realized is that I don't think we are modifying the right doc here. This is feature is really not about multi-tenancy at all, we only make sure here the right configuration is picked up for he offline token verification to succeed - but it can be noted in the example.

IMHO the doc updates belong to the `security-oidc-bearer-token-authentication.adoc`, I believe it would be the main flow for such use cases the token has arrived from somewhere, etc. Iif the practice shows it then we can link to the section in the bearer token guide from the code flow guide, but right now I believe it is not necessary.

Please move the docs, IMHO it would make it easier for the users who may need to find this feature in the bearer token concept doc